### ISSUE TITLE
Remove --with-mixedrefs configuration option

### ISSUE THREAD

**Issue by keithc-ca** [eclipse-openj9/openj9#18124]
All versions on all platforms have for some time been using mixed references; we no longer need the `--with-mixedrefs` configuration option (and its associated complexity).

**Comment by pshipton** [comment]
> `--with-mixedrefs=dynamic` create a mixed references build that uses runtime checks

Will all the runtime checks be removed as well?

**Comment by keithc-ca** [comment]
> Will all the runtime checks be removed as well?

No, some will remain, i.e. those places in the code where `J9VMTHREAD_COMPRESS_OBJECT_REFERENCES(thread)` or `J9JAVAVM_COMPRESS_OBJECT_REFERENCES(vm)` are not compile-time constants (e.g. in `runtime/vm/ContinuationHelpers.cpp`). The important parts of the VM are split (e.g. `BytecodeInterpreterCompressed.cpp` and `BytecodeInterpreterFull.cpp`): in those places those macros yield compile-time constants (as before).

The only thing we lost, is the `--with-mixedrefs=dynamic` case where _**none**_ of those uses of those macros yield compile-time constants (except perhaps for 32-bit builds). To my knowledge, nobody was using that configuration.

**Comment by pshipton** [comment]
It just seemed like there was more obsolete code to remove if dynamic mode can no longer be used.

**Comment by keithc-ca** [comment]
Most of the support for `OMR_MIXED_REFERENCES_MODE=dynamic` is in OMR. I'm not sure it's worth the effort to remove it just because OpenJ9 doesn't use it (and it's possible some other project downstream of OMR could be using it).

**Comment by pshipton** [comment]
@amicic @dmitripivkine fyi the recent changes to remove the OpenJ9 ability to use `--with-mixedrefs=dynamic`.

**Comment by keithc-ca** [comment]
Closing as complete via the pull requests linked above.
